internal/backend/schedule/scheduleworkflow.go

```diff
@@ -0,0 +1,150 @@
+	if !envID.IsValid() {
+		z.Debug("invalid environment")
+		return nil, fmt.Errorf("environment: %w", sdkerrors.ErrNotFound)
+	}
+
+	depFilter := sdkservices.ListDeploymentsFilter{State: sdktypes.DeploymentStateActive, EnvID: envID}
+	deployments, err := swf.Services.Deployments.List(ctx, depFilter)
+	if err != nil {
+		z.Panic("could not fetch active deployments", zap.Error(err))
+	}
+
+	if len(deployments) == 0 {
+		z.Debug("no deployments")
+		return nil, nil
+	}
+
+	cl := trigger.CodeLocation()
+	var sds []wf.SessionData
+	for _, dep := range deployments {
+		sds = append(sds, wf.SessionData{Deployment: dep, CodeLocation: cl, Trigger: trigger})
+		z.Debug("deployment found", zap.String("deployment_id", dep.ID().String()))
+	}
+
+	return sds, nil
+}
+
+func (swf *SchedulerWorkflow) newScheduleTickEvent(ctx context.Context, triggerID sdktypes.TriggerID) sdktypes.Event {
+	z := swf.Z.With(zap.String("trigger_id", triggerID.String()))
+
+	event := kittehs.Must1(sdktypes.EventFromProto(
+		&sdktypes.EventPB{
+			EventType:    sdktypes.SchedulerTickEventType,
+			Memo:         map[string]string{"trigger_id": triggerID.String()},
+			ConnectionId: fixtures.BuiltinSchedulerConnectionID.String(),
+		}))
+
+	eventID, err := swf.Services.Events.Save(ctx, event) // create event
+	if err != nil {
+		z.Panic("failed creating schedule tick event")
+	}
+
+	// REVIEW: re-fetching newly created event is costly (db). Right now we are recreating event without CreatedAt.
+	// other solution is to cause events.SaveEvent to return created event and not just ID
+	event = kittehs.Must1(sdktypes.EventFromProto(&sdktypes.EventPB{EventId: eventID.String(), EventType: sdktypes.SchedulerTickEventType}))
+
+	swf.CreateEventRecord(ctx, eventID, sdktypes.EventStateProcessing) // add `processing` event record
+	return event
+}
+
+func (swf *SchedulerWorkflow) scheduleWorkflowInternal(wfctx workflow.Context, ctx context.Context, triggerID sdktypes.TriggerID, tickEvent *sdktypes.Event) error {
+	z := swf.Z.With(zap.String("trigger_id", triggerID.String()))
+	z.Info("started")
+
+	sds, err := swf.getSessionData(ctx, triggerID)
+	if err != nil {
+		z.Error("get session data", zap.Error(err))
+		return err
+	}
+
+	// REVIEW: simplify/extract schedule's version of common:CreateSessionsForWorkflow or keep unified?
+	if err = swf.StartSessions(wfctx, *tickEvent, sds); err != nil {
+		z.Error("failed start session", zap.Error(err))
+		return fmt.Errorf("faield start session: %w", err)
+	}
+
+	z.Info("finished")
+	return err
+}
+
+func (swf *SchedulerWorkflow) scheduleWorkflow(wfctx workflow.Context, triggerID sdktypes.TriggerID) (*scheduleWorkflowOutput, error) {
+	var err error
+
+	logger := workflow.GetLogger(wfctx)
+	logger.Info("started scheduler workflow", "trigger_id", triggerID.String())
+
+	ctx := context.Background()
+	tickEvent := swf.newScheduleTickEvent(ctx, triggerID) // create tick event and add <processing> state
+
+	state := sdktypes.EventStateCompleted
```

Comment:
shouldn't this be Created?

Reply:
there is no `Created`, but there is `Processing` and then `completed/failure`.
newScheduleTickEvent creates both event and Processing state.
if scheduleWorkflowInternal finishes without error, then state = Completed otherwise Failure.